azure/durable_functions/orchestrator.py

```diff
@@ -49,57 +49,66 @@ def handle(self, context: DurableOrchestrationContext):
-                generation_state = self._generate_next(generation_state)
-
-        except StopIteration as sie:
-            orchestration_state = OrchestratorState(
-                is_done=True,
-                output=sie.value,
-                actions=self.durable_context.actions,
-                custom_status=self.durable_context.custom_status)
-        except Exception as e:
-            orchestration_state = OrchestratorState(
-                is_done=False,
-                output=None,  # Should have no output, after generation range
-                actions=self.durable_context.actions,
-                error=str(e),
-                custom_status=self.durable_context.custom_status)
+
+        else:
+            self.generator = fn_output
+            try:
+                generation_state = self._generate_next(None)
+
+                while not suspended:
+                    self._add_to_actions(generation_state)
+
+                    if should_suspend(generation_state):
+
+                        # The `is_done` field should be False here unless
+                        # `continue_as_new` was called. Therefore,
+                        # `will_continue_as_new` essentially "tracks"
+                        # whether or not the orchestration is done.
+                        orchestration_state = OrchestratorState(
+                            is_done=self.durable_context.will_continue_as_new,
+                            output=None,
+                            actions=self.durable_context.actions,
+                            custom_status=self.durable_context.custom_status)
+                        suspended = True
+                        continue
+
+                    if (isinstance(generation_state, Task)
+                        or isinstance(generation_state, TaskSet)) and (
+                            generation_state.is_faulted):
+                        generation_state = self.generator.throw(
+                            generation_state.exception)
+                        continue
+
+                    self._reset_timestamp()
+                    generation_state = self._generate_next(generation_state)
+
+            except StopIteration as sie:
+                orchestration_state = OrchestratorState(
+                    is_done=True,
+                    output=sie.value,
+                    actions=self.durable_context.actions,
+                    custom_status=self.durable_context.custom_status)
+            except Exception as e:
+                orchestration_state = OrchestratorState(
+                    is_done=False,
+                    output=None,  # Should have no output, after generation range
+                    actions=self.durable_context.actions,
+                    error=str(e),
+                    custom_status=self.durable_context.custom_status)
+
+        # No output if continue_as_new was called
+        if self.durable_context.will_continue_as_new:
```

Comment:
we do not set output if continue_as_new was called

Reply:
This is mostly for consistency. I'm trying to avoid continue_as_new from sometimes returning something and sometimes not. Just to avoid subtle bugs